src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -635,6 +654,12 @@ AdaptiveExecutor(CitusScanState *scanState)
 
 	ExecuteSubPlans(distributedPlan);
 
+	bool hasDependentJobs = HasDependentJobs(job);
+	if (hasDependentJobs)
+	{
```

Comment:
Just to keep this function clean, could the `if` be moved into `ExecuteDependentTasks`?

Reply:
We use `hasDependentJobs` boolean to clean schemas as well. So if since we want to call the function(hasDependentJobs) once it makes sense to keep it here I think. Though maybe we can execute dependent jobs after creating a distributed execution, and that maybe the function could be more clean. I will take a look at it

Reply:
makes sense, ok keeping it this way